### PR TITLE
Add onUpload/onDownload examples

### PR DESCRIPTION
**Pull request by andreyaksenov**
Added the `onUpload`/`onDownload` functions to existing examples for uploading/downloading files and referenced them in docs.

### CODE REVIEW

File: codeSnippets/snippets/client-upload-file/src/main/kotlin/com/example/Application.kt [hunk 1]
```diff
@@ -21,7 +23,13 @@ fun main() {
                     append(HttpHeaders.ContentDisposition, "filename=ktor_logo.png")
                 })
             }
-        )
+        ) {
+            onUpload { bytesSentTotal, contentLength ->
+                fixedRateTimer("timer", true, 0L, 50L) {
```
Comment (rsinukov) [inline codeSnippets/snippets/client-upload-file/src/main/kotlin/com/example/Application.kt:28]: Why is there a timer here? Will it be created on every upload progress change notification?

**Review by rsinukov — APPROVED** [review]

**Review by rsinukov — CHANGES_REQUESTED** [review]
Sorry, clicked the wrong button =)

File: codeSnippets/snippets/client-upload-file/src/main/kotlin/com/example/Application.kt (hunk 1 above)
Comment (andreyaksenov) [reply codeSnippets/snippets/client-upload-file/src/main/kotlin/com/example/Application.kt:28]: this is the issue.. thanks, will update

Comment (andreyaksenov) [reply codeSnippets/snippets/client-upload-file/src/main/kotlin/com/example/Application.kt]: removed `fixedRateTimer `  from both examples

**Review by rsinukov — APPROVED** [review]

**Pull request merged**